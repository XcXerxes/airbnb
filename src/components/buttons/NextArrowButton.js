import React, {Component} from 'react'
import colors from '../../styles/colors'
import PropTypes from 'prop-types'
import Icon from 'react-native-vector-icons/FontAwesome'
import {
  TouchableHighlight,
  StyleSheet
} from 'react-native'

export default class NextArrowButton extends Component {
  render () {
    const {disabled, handleNextButton} = this.props
    const opacityStyle = disabled ? {backgroundColor: 'rgba(255, 255, 255, .2)'} : {backgroundColor: 'rgba(255, 255, 255, .6)'} 
    return (
      <TouchableHighlight style={[opacityStyle, styles.button]} 
      disabled={disabled}
      onPress={handleNextButton}>
        <Icon name="angle-right" 
          color={colors.green01}
          size={32}
          style={styles.icon}
        />
      </TouchableHighlight>
    )
  }
}
NextArrowButton.propTypes = {
  disabled: PropTypes.bool,
  handleNextButton: PropTypes.func
}

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 50,
    width: 60,
    height: 60
  },
  icon: {
    marginRight: -2,
    marginTop: -2
  }
})
