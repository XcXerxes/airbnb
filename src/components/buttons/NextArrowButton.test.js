import React from 'react'
import NextArrowButton from './NextArrowButton'

import TestRender from 'react-test-renderer'

const testRender = TestRender.create(<NextArrowButton disabled={true} />)
const testInstance = testRender.root
const testTree = testRender.toJSON()


it('should renders components shot', () => {
  expect(testTree).toMatchSnapshot()
})

it('should renders props of disabled', () => {
  expect(testInstance.findByType(NextArrowButton).props.disabled).toBe(true)
})

it('should renders props number one', () => {
  expect(Object.values(testInstance.props).length).toBe(1)
})
