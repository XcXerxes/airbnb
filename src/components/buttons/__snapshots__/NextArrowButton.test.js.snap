// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders components shot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 255, 255, .2)",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 60,
        "justifyContent": "center",
        "width": 60,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
